src/Utilities/likelihood_informed.jl

```diff
@@ -0,0 +1,251 @@
+# included in Utilities.jl
+
+using Manifolds, Manopt
+
+export LikelihoodInformed, likelihood_informed
+
+"""
+$(TYPEDEF)
+
+Uses both input and output data to learn a subspace that allows for a reduced posterior which is close to the full posterior.
+
+Preferred construction is with the [`likelihood_informed`](@ref) method.
+
+# Fields
+$(TYPEDFIELDS)
+"""
+mutable struct LikelihoodInformed{FT <: Real} <: PairedDataContainerProcessor
```

Comment:
In general in the package we have never used mutable structs, only immutable ones. Which is perhaps annoying, but for consistency could we stick with immutable for now and do a larger change later (sorry)?

I also saw a couple of issues like this [ where the union {Nothing,X} do weird things before mutation](https://github.com/JuliaLang/julia/issues/58470). Not a big deal but happens quite commonly